hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -603,12 +605,24 @@ public FileStatusAdapter getFileStatus(String path, URI uri,
     if (ofsPath.isRoot()) {
       return getFileStatusAdapterForRoot(uri);
     }
+
+    OzoneVolume volume = objectStore.getVolume(ofsPath.getVolumeName());
```

Comment:
I'm guessing this line was moved here out of the if statement below because it is needed elsewhere.  That is often a good idea, but I think I remember people saying they were concerned about the performance of getFileStatus(), so I would leave it where it was and invoke it again from within the
```
if (ofsPath.isSnapshotPrefix()) {
```
block.